project-base/tests/App/Test/Codeception/Module/StrictWebDriver.php

```diff
@@ -154,10 +192,26 @@ public function fillField($field, $value)
      */
     public function fillFieldByElement(WebDriverElement $element, $value)
     {
-        $element->clear();
+        $this->moveMouseOverByElement($element);
+
+        $this->clearElement($element);
+
         $element->sendKeys($value);
     }
 
+    /**
+     * @param \Facebook\WebDriver\WebDriverElement $element
+     */
+    protected function clearElement(WebDriverElement $element): void
+    {
+        $element->clear();
+
+        while ($element->getAttribute('value') !== '') {
+            $element->sendKeys(WebDriverKeys::BACKSPACE);
+            $element->sendKeys(WebDriverKeys::DELETE);
+        }
```

Comment:
Is this necessary? I thought clearing should be done by `clear` method.

Reply:
Well it does not work in some cases, for example on our commerce cloud.

Reply:
interesting. I suggest a comment here then.

Reply:
What should it say?

Reply:
something like `workaround to ensure field being cleared due to ...` (I don't know why it's not working)